dspace-server-webapp/src/test/java/org/dspace/app/rest/CollectionGroupRestControllerIT.java

```diff
@@ -0,0 +1,2392 @@
+                                     .contentType(contentType))
+                        .andExpect(status().isUnprocessableEntity());
+
+        getClient(token).perform(get("/api/core/collections/" + collection.getID() + "/adminGroup"))
+                        .andExpect(status().isNoContent());
+    }
+
+    @Test
+    public void postCollectionAdminGroupCreateAdminGroupUnProcessablePermanent() throws Exception {
+
+        ObjectMapper mapper = new ObjectMapper();
+        GroupRest groupRest = new GroupRest();
+        groupRest.setPermanent(true);
+        MetadataRest metadataRest = new MetadataRest();
+        metadataRest.put("dc.description", new MetadataValueRest("testingDescription"));
+        metadataRest.put("dc.subject", new MetadataValueRest("testSubject"));
+
+        groupRest.setMetadata(metadataRest);
+
+        String token = getAuthToken(admin.getEmail(), password);
+
+        getClient(token).perform(post("/api/core/collections/" + collection.getID() + "/adminGroup")
+                                     .content(mapper.writeValueAsBytes(groupRest))
+                                     .contentType(contentType))
+                        .andExpect(status().isUnprocessableEntity());
+
+        getClient(token).perform(get("/api/core/collections/" + collection.getID() + "/adminGroup"))
+                        .andExpect(status().isNoContent());
+
+    }
+
+    @Test
+    public void deleteCollectionAdminGroupTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+        Group adminGroup = collectionService.createAdministrators(context, collection);
+        context.restoreAuthSystemState();
+
+        String token = getAuthToken(admin.getEmail(), password);
+        getClient(token).perform(delete("/api/core/collections/" + collection.getID() + "/adminGroup"))
+                        .andExpect(status().isNoContent());
+
+        getClient(token).perform(get("/api/core/collections/" + collection.getID() + "/adminGroup"))
+                        .andExpect(status().isNoContent());
+    }
+
+
+    @Test
+    public void deleteCollectionAdminGroupTestParentCommunityAdmin() throws Exception {
+        context.turnOffAuthorisationSystem();
+        Group adminGroup = collectionService.createAdministrators(context, collection);
+        authorizeService.addPolicy(context, parentCommunity, Constants.ADMIN, eperson);
+        context.restoreAuthSystemState();
+
+        String token = getAuthToken(eperson.getEmail(), password);
+        getClient(token).perform(delete("/api/core/collections/" + collection.getID() + "/adminGroup"))
+                        .andExpect(status().isNoContent());
+
+        getClient(token).perform(get("/api/core/collections/" + collection.getID() + "/adminGroup"))
+                        .andExpect(status().isNoContent());
+    }
+
+    // This is currently not supported in DSpace API
+    @Ignore
```

Comment:
I might be mistaken, but I think this `@Ignore` (and the comment above it) can also be removed now.

Reply:
@tdonohue No we cannot remove it from this specific test as it currently is. What this test tests is that a collection administrator can remove the admin group from a collection. If he was able to do this he would be removing access to himself (so this currently isn't allowed in the API).

What we can do, is that we alter the test to ensure that a collection admin cannot perform this action (which is what the API currently enforces) what do you think ?

Reply:
@KevinVdV : I'd recommend altering the test to ensure the Collection Admin *cannot* perform this action. That makes more sense as it's testing how the API currently works.  In the future if the behavior is changed, then this test can be changed.  Essentially, it's better to have tests the test the *current behavior* than to have `@Ignored` tests (which obviously never run, so they don't really test anything).